Rick&Morty/RickAndMorty/Common/DependencyInjection/StoreRegistration.swift

```diff
@@ -9,5 +9,15 @@
 import DependencyInjection
 
 @MainActor enum StoreRegistration {
-    static func registerDependencies(to container: Container) {}
+    static func registerDependencies(to container: Container) {
+        container.register(type: CharacterDetailStore.self) { _, character in
```

Comment:
Can you answer the question "why is here no scope whereas with the LocationsListStore we had to specify the new scope"?

Reply:
To answer this question - Yes, I can. 😃
`DependencyWithArgumentAutoregistering` uses automatically `new` scope, while `DependencyAutoregistering` doesn't (we have to select scope). You know the answer and also I - its in nice comments for protocols you wrote... 💪

If students ask this - simple answer would be that CharacterDetailStore is not created at the beginning when app starts/or when it is called - it will be created (as `new`) when we provide character (until than, it can't be created). 
LocationListStore doesn't have argument, so it can be created at the beginning of app. We can select new/shared but in this case we need `new` - bc it will hold data (such as state in next lecture).

Reply:
I just challenged you 🙂 Perfect 👏